lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -412,7 +412,15 @@ def cleanup():
             "data" in deployment_desc.config.roles
             and self.unit.is_leader()
             and deployment_desc.typ == DeploymentType.OTHER
-            and not self.peers_data.get(Scope.APP, "security_index_initialised", False)
+            and (
```

Comment:
This is unrelated to this line, but just caught it. We are missing a check on line 390. Currently, this logic leaves us vulnerable to [LP#2076599](https://launchpad.net/bugs/2076599). Can we have instead:
```
if not (deployment_desc := self.opensearch_peer_cm.deployment_desc()):
    event.defer()
    return
```

Reply:
No need to check this here, is already covered in line 359 with `_apply_peer_cm_directives_and_check_if_can_start()`, which checks if the `deployment_desc` is available.